auth/domain/src/test/java/com/pronoidsoftware/auth/domain/UserDataValidatorTest.kt

```diff
@@ -0,0 +1,90 @@
+package com.pronoidsoftware.auth.domain
+
+import assertk.assertThat
+import assertk.assertions.isFalse
+import assertk.assertions.isTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+internal class UserDataValidatorTest {
+
+    private lateinit var sut: UserDataValidator
+    private lateinit var emailPatternValidatorFake: PatternValidator
+
+    @BeforeEach
+    fun setUp() {
+        // we don't need to check that the email pattern matcher works, so returning true
+        emailPatternValidatorFake = object : PatternValidator {
+            override fun matches(value: String): Boolean {
+                return true
+            }
+        }
```

Comment:
Strictly speaking not a fake, but a stub 😄 Fakes simulate the real behavior which this object doesn't do, since it would also return true for invalid inputs. Stubs just return fix sample data.